incubator/hnc/e2e/tests/delete_crd_test.go

```diff
@@ -0,0 +1,41 @@
+package test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("Delete-anchor-crd", func() {
+	hncRecoverPath := os.Getenv("HNC_REPAIR")
+	AfterEach(func() {
+		// clean up
+		nses := []string{
+			"delete-crd-parent",
+			"delete-crd-child",
+		}
+		// Remove all possible objections HNC might have to deleting a network. Make sure it 
+		// has cascading deletion so we can delete any of its subnamespace descendants, and 
+		// make sure that it's not a subnamespace itself so we can delete it directly.
+		for _, ns := range nses {
+			tryRun("kubectl hns set", ns, "-a")
+			tryRun("kubectl annotate ns", ns, "hnc.x-k8s.io/subnamespaceOf-")
+			tryRun("kubectl delete ns", ns)
+		}
+		mustRun("kubectl apply -f", hncRecoverPath)
```

Comment:
@GinnyJI Can you please file an issue to make this more robust? I think I know what you mean - that the recovery "has" to work - but mustRun will only stop the current _test_ from running if it fails, not the current _test suite_. We might want to do something more like:

```
if err := tryRun( ... ); err != nil {
  GinkoT().Logf("Couldn't repair HNC: %v", err)
  os.Exit(1)
}
```

That will stop all _other_ tests from running, which is probably what we want here.

Reply:
Done at issue #881